tools/util/half.h

```diff
@@ -229,7 +235,10 @@ int fpclassify(cutlass::half_t const&);  /// returns a flag classifying floating
 bool signbit(cutlass::half_t const&);  /// returns true if negative, false if positive
 
 cutlass::half_t sqrt(cutlass::half_t const&);  /// square root of half_t
+cutlass::half_t copysign(cutlass::half_t const&, cutlass::half_t const&);
```

Comment:
I guess this is another c++11 feature?

Reply:
No, it's a forward declaration for the function already implemented at the end of this header. Unlike the sqrt() and other functions above, this one wasn't declared.

I've added the declaration for consistency. AFAICT, it's not strictly necessary and I can remove it, if you prefer.

Reply:
When I compile the code in this MR with GCC without C++11. It did complain `error: namespace "std" has no member "copysign"`. If you think copysign is not needed or not necessary could we remove it?

Reply:
Scratch my previous comment. I did add both the declaration and the implementation. I'll need to check tomorrow, but I believe it was needed to make the standard library happy.  I'll check if it's still needed. If it is, I can preprocess it away for C++ standards older than c++11. Will that be acceptable?

Reply:
hehe. Yes I think that's reasonable.  
It looks like the code as is makes call to std::copysign
```
inline cutlass::half_t copysign(cutlass::half_t const& a,
                                cutlass::half_t const& b) {
  return cutlass::half_t(std::copysign(float(a), float(b)));
}
```
which is a C++11 feature. But I cannot seem to find the code that makes call to it. Anyways, we would like to make sure cutlass 1.3 does not use c++11 feature. I think it is okay for clang compiler to use `--std=c++11` But we want to make sure cutlass can still compile with older compiler that does not support cpp11.

Reply:
Hi, Artem!

Specifically, this is a library feature added in C++11. I don't think this depends on any actual C++11 language features, so it's okay to use.

I'd recommend contributing this function and others like it to `cutlass/util/platform.h` so we can swap it out at a point in the future when we can fully embrace C++11.

Reply:
On second thought, quite a few functions above are only in c++11 (e.g. [signbit](https://en.cppreference.com/w/cpp/numeric/math/signbit), [isnan](https://en.cppreference.com/w/cpp/numeric/math/isnan), [isfinite](https://en.cppreference.com/w/cpp/numeric/math/isfinite))

I'm not sure it's worth making a special case just for `copysign.

Reply:
@TimmyLiu  makes a valid point that I'm also using `std::copysign` which may not be available before c++11. That indeed breaks if nvcc compiles the files using -std=c++03. I've added an  `#if __cplusplus` guard around it.

Reply:
Thank you!